### PR TITLE
meta state

### PR DESCRIPTION
**Pull request by phantomcosmonaut**
Add `setMetaState` reducer
Extend slice reducers from meta slice reducers
remove unnecessary typings

### CODE REVIEW

File: packages/redux-slice-factory/src/slice-factories/meta-slice/MetaSlice.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+     /**
+      * This will set the error of the slice.
+      */
+     setError: CaseReducer<TSliceState, PayloadAction<TError | null>>;
+ 
+     /**
+      * This will set any meta state properties that are defined
+      */
+     setMetaState: CaseReducer<TSliceState, PayloadAction<Partial<IMetaState<TStatusEnum, TError>>>>;
+}
```
Comment (gregjoeval) [inline packages/redux-slice-factory/src/slice-factories/meta-slice/MetaSlice.ts:19]: this guy should go in types/index.ts

**Comment by phantomcosmonaut** [comment]
@gregjoeval kept meaning to get back to this pr. I've made the changes you requested and fixed the linting issues.

File: packages/redux-slice-factory/src/slice-factories/meta-slice/MetaSlice.ts (hunk 1 above)
Comment (phantomcosmonaut) [reply packages/redux-slice-factory/src/slice-factories/meta-slice/MetaSlice.ts]: done

**Comment by phantomcosmonaut** [comment]
@gregjoeval I ran the tests locally and they passed, not sure how to handle the build failure

**Comment by gregjoeval** [comment]
> @gregjoeval I ran the tests locally and they passed, not sure how to handle the build failure

I'll take a look at this either tonight or tomorrow. Apologies for the delays, work's been pretty busy as of late. Thanks for the work so far!

**Comment by phantomcosmonaut** [comment]
@gregjoeval Mind taking another look at this?

**Comment by gregjoeval** [comment]
i believe its an issue with `createSelector`. There was an update to that function within the past few months that changed how its types worked. I dont have the time to figure it out right now but ill make time to look into it soon.

**Comment by gregjoeval** [comment]
i got really tired of trying to satisfy the very ornery type constraints of the output selector, so for the time being the type assertions will be fine, they are technically correct. 

The typescript extension on my personally machine either wasn't working or wasnt picking up the same type errors as the build script which added another layer of [FUN](https://dwarffortresswiki.org/index.php?title=DF2014:Fun&redirect=no)

@phantomcosmonaut this should be good to go now, apologies for the long wait, been busy.

**Review by gregjoeval — APPROVED** [review]

**Comment by gregjoeval** [comment]
I will publish this as a new minor tonight or tomorrow

**Pull request merged**